pyfr/plugins/fluidforce.py

```diff
@@ -9,46 +9,20 @@
 
 
 class FluidForceIntegrator(SurfaceIntegrator):
-    def __init__(self, cfg, cfgsect, system, bcname, viscous, morigin):
+    def __init__(self, cfg, cfgsect, system, bcname, morigin):
         surf_list = system.mesh.bcon.get(bcname, [])
         surf_list = [(etype, fidx, eidxs) for etype, eidxs, fidx in surf_list]
-        
-        super().__init__(cfg, cfgsect, system.ele_map, surf_list, flags='s')
-
-        if viscous:
-            self.m4 = m4 = {}
-            rcpjact = {}
-
-        if surf_list:
-            rfpts = defaultdict(list)
-            for etype, fidx in self.eidxs:
-                for i, eidx in enumerate(self.eidxs[etype, fidx]):
-                    eles = system.ele_map[etype]
-
-                    if viscous and etype not in m4:
-                        m4[etype] = eles.basis.m4
-
-                        # Get the smats at the solution points
-                        smat = eles.smat_at_np('upts').transpose(2, 0, 1, 3)
-
-                        # Get |J|^-1 at the solution points
-                        rcpdjac = eles.rcpdjac_at_np('upts')
 
-                        # Product to give J^-T at the solution points
-                        rcpjact[etype] = smat*rcpdjac
-
-                    # Get the flux points position of the given face and 
-                    # element indices relative to the moment origin
-                    if morigin is not None:
-                        ploc = self.locs[etype, fidx][i]
-                        rfpt = ploc - morigin
-                        rfpts[etype, fidx].append(rfpt)
-        
-            self.rfpts = {k: np.array(v) for k, v in rfpts.items()}
-            if viscous:
-                self.rcpjact = {k: rcpjact[k[0]][..., v]
-                                for k, v in self.eidxs.items()}
+        super().__init__(cfg, cfgsect, system.ele_map, surf_list, flags='s')
 
+        if surf_list and morigin is not None:
+            self.rfpts = {
+                (etype, fidx): np.array([
+                    self.locs[etype, fidx][i] - morigin
+                    for i, _ in enumerate(self.eidxs[etype, fidx])
```

Comment:
I think this loop can be simplified if we loop over locs instead and then avoid looping over eidx's entirely.